### PR TITLE
Upgrade to graphql-java 19.0 & Federation 2.1.0

### PR DESCRIPTION
**Pull request by berngp**
## Pull Request type
----

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] Other (please describe):

### GraphQL

GraphQL dependency changes...
* "com.graphql-java:graphql-java" to "19.2"
* "com.graphql-java:graphql-java-extended-scalars"i to "19.0"
* "com.graphql-java:graphql-java-extended-validation" to "19.1-hibernate-validator-6.2.0.Final"
* "com.apollographql.federation:federation-graphql-java-support" to "2.1.0"

### Kotlin 1.7.20

With the intention to keep up with the latest version of Kotlin, we are upgrading to 1.7.20. Migration to Kotlin 1.7 should be fairly simple. Please review ["What's new in Kotlin 1.7.0"](https://kotlinlang.org/docs/whatsnew17.html) for further reference.

### Spring Boot 2.6

Latest Spring Boot 2.6, this includes upgrading...
* "org.springframework:spring-framework-bom:5.3.23"
* "org.springframework.security:spring-security-bom" to "5.7.3"
* "com.fasterxml.jackson:jackson-bom" to "2.13.4"

### Other

* "com.jayway.jsonpath:json-path" to "2.7.0"
* "io.projectreactor:reactor-core" to "3.4.22"
* "io.projectreactor:reactor-test" to "3.4.22"

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -28,7 +28,7 @@ plugins {
     `java-library`
     id("nebula.dependency-recommender") version "11.0.0"
     id("nebula.netflixoss") version "10.6.0"
-    id("org.jmailen.kotlinter") version "3.10.0"
+    id("org.jmailen.kotlinter") version "3.11.1"
```
Comment (berngp) [inline build.gradle.kts:31]: Most of the KT changes were made via the 3.11 version of kotlinter.

File: build.gradle.kts [hunk 2]
```diff
@@ -50,16 +50,16 @@ allprojects {
...
     // in buildSrc, most likley because the variables are used in plugins as well
     // as dependencies. e.g. KOTLIN_VERSION
-    extra["sb.version"] = "2.6.7"
+    extra["sb.version"] = "2.6.12"
     val springBootVersion = extra["sb.version"] as String
 
     dependencyRecommendations {
         mavenBom(mapOf("module" to "org.jetbrains.kotlin:kotlin-bom:${Versions.KOTLIN_VERSION}"))
-        mavenBom(mapOf("module" to "org.springframework:spring-framework-bom:5.3.18"))
+        mavenBom(mapOf("module" to "org.springframework:spring-framework-bom:5.3.23"))
```
Comment (berngp) [inline build.gradle.kts:58]: Changes to `spring-framework-bom`, `spring-security-bom`, and `jackson-bom` align with the versions used in SB 2.6.12.

File: graphql-dgs-example-shared/src/test/java/com/netflix/graphql/dgs/example/shared/HelloDataFetcherTest.java [hunk 3]
```diff
@@ -56,8 +61,9 @@ void getQueryWithInspectError() {
             queryExecutor.executeAndExtractJsonPath("{greeting}", "data.greeting");
             fail("Exception should have been thrown");
         } catch (QueryException ex) {
-            assertThat(ex.getMessage()).contains("Validation error of type FieldUndefined: Field 'greeting' in type 'Query' is undefined @ 'greeting'");
-            assertThat(ex.getErrors().size()).isEqualTo(1);
+            assertThat(ex.getMessage())
+                    .contains("Validation error (FieldUndefined@[greeting]) : Field 'greeting' in type 'Query' is undefined");
```
Comment (berngp) [inline graphql-dgs-example-shared/src/test/java/com/netflix/graphql/dgs/example/shared/HelloDataFetcherTest.java:65]: Error format changed in graphql-java 19.

File: graphql-dgs-platform/build.gradle.kts [hunk 4]
```diff
@@ -48,27 +48,28 @@ dependencies {
...
+            version { require("19.2") }
         }
         api("com.graphql-java:graphql-java-extended-scalars") {
-            version { require("18.1") }
+            version { require("19.0") }
         }
         api("com.graphql-java:graphql-java-extended-validation") {
             // The version below will work with Jakarta EE 8 and use Hibernate Validator 6.2.
+            // It is meant to work on graphql-java 18.x, that said there is no version for it yet.
             version { require("18.1-hibernate-validator-6.2.0.Final") }
```
Comment (berngp) [inline graphql-dgs-platform/build.gradle.kts:59]: There is no `19.0-hibernate-validator-6.2.0.Final` version, nor I think there will be.

**Comment by setchy** [comment]
Awesome @berngp 👏 

re: `federation-jvm`, there was a [2.1.0 release](https://github.com/apollographql/federation-jvm/releases/tag/v2.1.0) late yesterday

re `graphql-java-extended-validation`, would the [v19.1-validator-6.2.0.Final](https://github.com/graphql-java/graphql-java-extended-validation/releases/tag/v19.1-validator-6.2.0.Final) release be worthy of updating to?

File: graphql-dgs-platform/build.gradle.kts (hunk 4 above)
Comment (berngp) [reply graphql-dgs-platform/build.gradle.kts:59]: Thanks to @setchy for pointing out that there is a `v19.1-validator-6.2.0.Final`! Will upgrade.

**Comment by berngp** [comment]
Thanks @setchy, bumped both versions after you made me aware of the availability.

**Review by srinivasankavitha — APPROVED** [review]

**Pull request merged**